src/lib/components/tablev2/MultiSelectableContent.tsx

```diff
@@ -0,0 +1,281 @@
+import { useEffect, memo, useCallback, useState, CSSProperties } from 'react';
+import { Row } from 'react-table';
+import { areEqual } from 'react-window';
+
+import { ConstrainedText } from '../constrainedtext/Constrainedtext.component';
+import { Tooltip } from '../tooltip/Tooltip.component';
+import { useTableContext } from './Tablev2.component';
+import {
+  HeadRow,
+  NoResult,
+  SortCaretWrapper,
+  SortIncentive,
+  TableBody,
+  TableHeader,
+  TableRowMultiSelectable,
+  TooltipContent,
+  UnknownIcon,
+} from './Tablestyle';
+import {
+  TableHeightKeyType,
+  TableLocalType,
+  TableVariantType,
+  VirtualizedRows,
+} from './TableUtils';
+
+const translations = {
+  en: {
+    noResult: 'No results found',
+  },
+  fr: {
+    noResult: `Aucun résultat`,
+  },
+};
+
+type RenderRowType = {
+  index: number;
+  style: CSSProperties;
+};
+
+type MultiSelectableContentProps = {
+  onMultiSelectionChanged: (rows: Row<object>[]) => void;
+  rowHeight?: TableHeightKeyType;
+  separationLineVariant?: TableVariantType;
+  backgroundVariant?: TableVariantType;
+  locale?: TableLocalType;
+  customItemKey?: (index: Number, data: any) => string;
+  children?: (rows: JSX.Element) => JSX.Element;
+};
+
+export const MultiSelectableContent = ({
+  onMultiSelectionChanged,
+  rowHeight = 'h40',
+  separationLineVariant = 'backgroundLevel3',
+  backgroundVariant = 'backgroundLevel1',
+  locale = 'en',
+  customItemKey,
+  children,
+}: MultiSelectableContentProps) => {
+  const [hasScrollbar, setHasScrollbar] = useState(false);
+  const [scrollBarWidth, setScrollBarWidth] = useState(0);
+
+  const {
+    headerGroups,
+    prepareRow,
+    rows,
+    setHiddenColumns,
+    selectedRowIds,
+    onBottom,
+    onBottomOffset,
+    isAllRowsSelected,
+  } = useTableContext();
+
+  useEffect(() => {
+    setHiddenColumns([]);
+  }, [setHiddenColumns]);
```

Comment:
For now this is ok but maybe later on we will have use cases with hidden columns, So I wonder wether we could not jsut remove the checkbox column from the hidden columns instead of forcing to an empty array here.
Something like this maybe:

```suggestion
  useEffect(() => {
    setHiddenColumns((oldHiddenColumns) => oldHiddenColumns.filter(column => column.id !== 'selection'));
  }, [setHiddenColumns]);
```